src/UserAction/StepExecutorTrait.php

```diff
@@ -166,8 +189,12 @@ protected function doArgs(View $page): void
         $this->jsSetPrevHandler($page, $this->step);
 
         $form->onSubmit(function (Form $form) {
+            $form->model->save();
             // collect arguments.
-            $this->actionData['args'] = $form->model->get();
+            foreach ($form->model->getFields('editable') as $k => $field) {
+                $argValues[$k] = $form->model->get($k);
+            }
+            $this->setActionData('args', $argValues ?? []);
```

Comment:
```suggestion
            $argValues = [];
            foreach ($form->model->getFields('editable') as $k => $field) {
                $argValues[$k] = $form->model->get($k);
            }
            $this->setActionData('args', $argValues);
```

never use `??` for possibly undefined local variables, use it only for possibly undefined keys, it is otherwise hard to read

Reply:
done